src/consensus/params.h

```diff
@@ -25,6 +37,10 @@ struct Params {
     int64_t nPowTargetSpacing;
     int64_t nPowTargetTimespan;
     int64_t DifficultyAdjustmentInterval() const { return nPowTargetTimespan / nPowTargetSpacing; }
+    uint64_t MaxBlockSize(uint64_t nCurrentTime) const { return blockParams->MaxBlockSize(nCurrentTime); }
```

Comment:
In https://github.com/jtimon/bitcoin/commit/c4cfb5e098d23e017f98aad93698350bb91960b4 the functions are equal for all chains (just like any other consensus function), but the values they take in Consensus::Params are different. You don't need polymorphism for this: you can add as many attributes as you need in Consensus::Params (even bool ones) and sets their values appropriately.
For example, 
```
uint64_t MaxBlockSize(const Consensu::Params& consensusParams, uint64_t nCurrentTime) const
{
        // Increases 1000000 bytes/day starting at the regtest Genesis Block's
        // timestamp.
        if (nCurrentTime >= consensusParams.nStartTime + consensusParams.nDay)
            return consensusParams.nBaseSize + consensusParams.nStep * ((nCurrentTime - consensusParams.nStartTime) / consensusParams.nDay);
        return consensusParams.nBaseSize;
}
```
Just set consensusParams.nStartTime to max<uint64_t> for all the chains but the testing one and you are done.
See https://github.com/jtimon/bitcoin/commit/06872b568e9b74fdce3d6e8ff0a15ac3095074d4 too.